Controller/CaptureController.php

```diff
@@ -19,12 +22,22 @@ public function doSessionTokenAction(Request $request)
 
         $request->getSession()->remove('payum_token');
 
-        return $this->redirect($this->generateUrl('payum_capture_do', array_replace(
+        $redirectUrl = $this->generateUrl('payum_capture_do', array_replace(
             $request->query->all(),
             array(
                 'payum_token' => $hash,
             )
-        )));
+        ));
+
+        if ($request->isMethod('GET')) {
+            return $this->redirect($redirectUrl);
+        }
+
+        try {
+            throw new HttpPostRedirect($redirectUrl, $request->request->all());
```

Comment:
There is a listener that could convert reply to the response. Could we use it?

Reply:
Done, I didn't know it was listening all controller actions!